### PR TITLE
Update sommelier to 20221117

### PR DESCRIPTION
**Pull request by satmandu**
- Updated to the 20221117 version @supechicken noticed.
- Cleaned up output and error messages.

Works properly:
- [x] `x86_64`

Builds properly:
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=sommelier_new CREW_TESTING=1 crew update
```

### CODE REVIEW

File: packages/sommelier.rb [hunk 1]
```diff
@@ -75,10 +75,9 @@ def self.build
           -Db_asneeded=false \
           -Db_lto=true \
           -Db_lto_mode=thin \
+          -Dwith_tests=false \
           -Dxwayland_path=#{CREW_PREFIX}/bin/Xwayland \
           -Dxwayland_gl_driver_path=/usr/#{ARCH_LIB}/dri -Ddefault_library=both \
```
Comment (supechicken) [inline packages/sommelier.rb:80]: Use the `crew` version of `mesa`? (some ARM Chromebooks (like mine) do not come across with the `mesa` drivers preinstalled, all drivers are merged into the chrome binary instead)

File: packages/sommelier.rb [hunk 2]
```diff
@@ -200,15 +199,17 @@ def self.build
...
 
             sommelier -X \
-              --x-display=${DISPLAY}  \
-              --scale=${SCALE} --glamor \
-              --drm-device=/dev/dri/renderD128 \
-              --virtwl-device=/dev/null \
-              --shm-driver=noop --data-driver=noop \
+              --x-display=${DISPLAY} \
+              --scale=${SCALE} \
+              --direct-scale \
```
Comment (supechicken) [inline packages/sommelier.rb:204]: Notice the `--direct-scale` option only works on recent version of Chrome OS:
```
# on my ARM chromebook, chrome 84
sommelier.elf: ../sommelier-output.cc:180: void sl_output_init_dimensions_direct(struct sl_host_output *, int *, int *, int *, int *, int *): Assertion `host->ctx->xdg_output_manager' failed.
```

**Comment by supechicken** [comment]
Works on `x86_64` (`qemu+virgl` and intel hardware), **with `/dev/dri/renderD129` as DRM device**:

Failed on my ARM Chromebook, this patch I figured out last week fixed it:
```
sommelier.elf: ../sommelier-window.cc:412: void sl_window_update(struct sl_window *): Assertion `ctx->xdg_shell' failed.
(EE) failed to write to Xwayland fd: Broken pipe
Aborted (core dumped)
```

```patch
diff -Nur a/sommelier.cc b/sommelier.cc
--- a/sommelier.cc	2022-11-06 19:29:16.580361574 +0800
+++ b/sommelier.cc	2022-11-06 19:37:28.830367176 +0800
@@ -616,7 +616,7 @@
       data_device_manager->host_global =
           sl_data_device_manager_global_create(ctx);
     }
-  } else if (strcmp(interface, "xdg_wm_base") == 0) {
+  } else if ((strcmp(interface, "xdg_wm_base") == 0) || (strcmp(interface, "zxdg_shell_v6") == 0)) {
     struct sl_xdg_shell* xdg_shell =
         static_cast<sl_xdg_shell*>(malloc(sizeof(struct sl_xdg_shell)));
     assert(xdg_shell);
```

**Comment by supechicken** [comment]
Move changes to `crew-package-sommelier` for better maintenance?

**Comment by satmandu** [comment]
@supechicken do you want to hack this up to get it to the finish line?

File: packages/sommelier.rb (hunk 2 above)
Comment (satmandu) [reply packages/sommelier.rb:204]: Do you know which ChromeOS Milestone this starts working with? And is that a fatal error on the older versions?
Any ideas for a milestone version check we can use in the sommelierd script to select whether to use `--direct-scale`?

File: packages/sommelier.rb (hunk 1 above)
Comment (satmandu) [reply packages/sommelier.rb:80]: Ah yes, I suppose that makes sense!

**Comment by satmandu** [comment]
Rebuilt this. @supechicken @uberhacker Would you please see if this works now on arm for you?

**Review by uberhacker — CHANGES_REQUESTED** [review]
Please remove the `--direct-scale` option.  I tested this on my armv7l system (stuck at release 84 like @supechicken) and it started working with that one change.

File: packages/sommelier.rb [hunk 3]
```diff
@@ -283,7 +308,7 @@ def self.build
           }
           if ! checksommelierwayland || ! checksommelierxwayland ; then
             [ -f  #{CREW_PREFIX}/bin/stopbroadway ] && stopbroadway
-            #{CREW_PREFIX}/sbin/sommelierd &
+            #{CREW_PREFIX}/sbin/sommelierd &>/dev/null
```
Comment (uberhacker) [inline packages/sommelier.rb:311]: What is wrong with seeing the output?

**Review by uberhacker — COMMENTED** [review]
Do we need new binaries?

File: packages/sommelier.rb (hunk 3 above)
Comment (satmandu) [reply packages/sommelier.rb:312]: The output is already being put into a log file from inside `sommelierd`

Comment (satmandu) [reply packages/sommelier.rb:312]:
```
            sommelier -X \
              --x-display=${DISPLAY} \
              --scale=${SCALE} \
              ${SOMMELIER_DIRECT_SCALE} \
              --glamor \
              --force-drm-device=${SOMMELIER_DRM_DEVICE} \
              --display=wayland-0 \
              --xwayland-path=/usr/local/bin/Xwayland \
              --xwayland-gl-driver-path=#{CREW_LIB_PREFIX}/dri \
              --peer-cmd-prefix="#{CREW_PREFIX}#{@peer_cmd_prefix}" \
              --enable-xshape \
              --noop-driver \
              --no-exit-with-child \
              /bin/sh -c "touch ~/.Xauthority
                xauth -f ~/.Xauthority add ${DISPLAY} . $(xxd -l 16 -p /dev/urandom)
                source #{CREW_PREFIX}/etc/sommelierrc" \
            &>> #{CREW_PREFIX}/var/log/sommelier.log
```

File: packages/sommelier.rb (hunk 2 above)
Comment (satmandu) [reply packages/sommelier.rb]: ok I inserted code to handle this.

**Comment by satmandu** [comment]
Rebuilt and added new binaries.

Please check with 
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=sommelier_new CREW_TESTING=1 crew update ; crew reinstall sommelier
```

File: packages/sommelier.rb (hunk 3 above)
Comment (uberhacker) [reply packages/sommelier.rb:312]: Gotcha.

**Review by uberhacker — APPROVED** [review]

**Comment by myelin** [comment]
Nice spotting... I was wondering if I should ping someone at Chromebrew to see if you were interested in switching to the newer Sommelier, but it looks like @supechicken found it almost immediately after it landed.

I'm curious about the zxdg_shell_v6 change; it looks like support for this was dropped in https://crrev.com/c/3205214 in Oct 2021, because it broke apps on Debian Stretch.  @supechicken what software were you running that caused Sommelier to crash with the ctx->xdg_shell assertion failure?

**Comment by supechicken** [comment]
I found your commit while tracing which commit breaks the noop driver patch (due to the new `/*is_drm=*/true` option in `sl_create_host_buffer()`). Anyways thank you for re-adding the noop driver to sommelier :)

Without modifying the `strcmp(interface, "xdg_wm_base")` line, every X program that requires opening a window triggers the `assert(ctx->xdg_shell)` line (commands that do not require creating an X window like `xdpyinfo` and `glxinfo -B` works well):
```
$ sommelier -X --noop-driver glxgears
sommelier.elf: ../sommelier-window.cc:412: void sl_window_update(struct sl_window *): Assertion `ctx->xdg_shell' failed.
(EE) failed to write to Xwayland fd: Broken pipe
Aborted (core dumped)
```

I tried adding `printf("%s", interface);` before the if statement [in `sommelier.cc`, line 619](https://chromium.googlesource.com/chromiumos/platform2/+/main/vm_tools/sommelier/sommelier.cc#619), it prints `zxdg_shell_v6` instead of the v7 `xdg_wm_base`

It seems that the exo server on CrOS v84 was compiled before https://crrev.com/c/2237428 was merged, and thus using the older v6 interface (surprisingly sommelier works without replacing the new `xdg_shell.xml` with the old v6_unstable one)

**Comment by satmandu** [comment]
Honestly the other lingering issue with the exo server is that `wl_compositor` version is still `3` instead of `4`. This is continuing to be an issue with building software which can run directly w/o sommelier. e.g. [mpv](https://github.com/mpv-player/mpv/issues/10868#issuecomment-1314584031).

**Pull request merged**